src/texstudio.cpp

```diff
@@ -710,8 +710,22 @@ void Texstudio::setupDockWidgets()
 
     leftPanel->showWidgets();
     // restore selected view in sidepanel
-    int viewNr=configManager.getOption("GUI/sidePanel/currentPage", 0).toInt();
-    leftPanel->setCurrentWidget(leftPanel->widget(viewNr));
+	QList<QString> hiddenWidgetsIdsList = leftPanel->hiddenWidgets().split("|");
+    int viewNr = configManager.getOption("GUI/sidePanel/currentPage", 0).toInt();
+	int k = -1; // index of visible tool found
+	for (int i = 0; i < leftPanel->widgetCount(); i++)  {
+		QString currentWidgetId = leftPanel->widget(i)->property("id").toString();
+		if (!hiddenWidgetsIdsList.contains(currentWidgetId)) {
+			k++;
+			if (k == viewNr) {
+				leftPanel->setCurrentWidget(leftPanel->widget(i));
+				emit leftPanel->titleChanged(leftPanel->widget(i)->property("Name").toString());
+				break;
+			}
+		}
+	}
+	if (k == -1) // there are no visible tools
```

Comment:
I guess, that now it is possible that viewNr is larger than the number of visible widgets (i.e. not widget selected) and k is already increased. Hence line 727 is not true (but should)

Reply:
that should be the last fix.

Reply:
How do you activate a tool with an index viewNr higher than the highest index of the visible tools? Anyway, to do so I testet what happens when I change the number to 20 (which is higher than the numer of availabel tools) in the ini file and start txs. Works fine. Why? Because in this case the initial configuration stays in effect. As you can see from Method showWidgets() this defaults (correctly) to the first visible tool (s.f. variable foundFirstNonHiddenWidget) or none (l. 559). So no fix from my side. One note: There where fixes and build in error (CI, CD) due to QList. I didn't see those errors in my compiles that I did before sending updates. I noticed that linux build runs with  make -j 2, so I changet to that from make -j 8. The latter is given in the wiki, so maybe this should be changed there (or make a note there)?

Reply:
`make -j n` just specifies the number of parallel build processes. It does not change the result.